powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/persistence/OperationQueryServiceDefault.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.service.persistence;
+
+import io.getlime.security.powerauth.app.server.configuration.conditions.IsNotMssqlCondition;
+import io.getlime.security.powerauth.app.server.database.model.entity.OperationEntity;
+import io.getlime.security.powerauth.app.server.database.repository.OperationRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * Service for operation queries with pessimistic locking.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Service
+@Conditional(IsNotMssqlCondition.class)
```

Comment:
🤔 Have you considered doing this conditional configuration even on the repository level and keeping a single service?

Reply:
I have considered it. If we do this for operations, we should do it for activations, too. However there will be many duplicate methods on `ActivationRepository`.